qiskit/algorithms/time_evolvers/classical_methods/utils.py

```
@@ -0,0 +1,181 @@
+
+    Args:
+        evolution_problem: The definition of the evolution problem.
+        steps: Number of timesteps to be performed.
+        real_time: If `True`, returned operators will correspond to real time evolution,
+            Else, they will correspond to imaginary time evolution.
+
+    Returns:
+        A tuple with the initial state, the list of operators to evaluate and the operator to be
+        exponentiated to perform one timestep.
+    """
+    # Convert the initial state and Hamiltonian into sparse matrices.
+    if isinstance(evolution_problem.initial_state, QuantumCircuit):
+        state = Statevector(evolution_problem.initial_state).data.T
+    else:
+        state = evolution_problem.initial_state.to_matrix(massive=True).transpose()
+
+    hamiltonian = evolution_problem.hamiltonian.to_spmatrix()
+
+    if isinstance(evolution_problem.aux_operators, list):
+        aux_ops = [aux_op.to_spmatrix() for aux_op in evolution_problem.aux_operators]
+    elif isinstance(evolution_problem.aux_operators, dict):
+        aux_ops = [aux_op.to_spmatrix() for aux_op in evolution_problem.aux_operators.values()]
+    else:
+        aux_ops = []
+    timestep = evolution_problem.time / steps
+    step_operator = -((1.0j) ** real_time) * timestep * hamiltonian
+    return state, aux_ops, step_operator
+
+
+def _evolve(
+    evolution_problem: TimeEvolutionProblem, steps: int, real_time: bool
+) -> TimeEvolutionResult:
+    r"""Performs either real  or imaginary time evolution :math:`\exp(-i t H)|\Psi\rangle`.
+
+    Args:
+        evolution_problem: The definition of the evolution problem.
+        steps: Number of timesteps to be performed.
+        real_time: If `True`, returned operators will correspond to real time evolution,
+            Else, they will correspond to imaginary time evolution.
+
+    Returns:
+        Evolution result which includes an evolved quantum state.
+
+    Raises:
+        ValueError: If the Hamiltonian is time dependent.
```

We also need an initial state, so we should probably also raise an error if that's not provided 🙂